<a href="https://colab.research.google.com/github/brunomsamorim/BlueTurma2B-modulo1/blob/main/Aula04_Exercicios.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exercícios de Aquecimento

## Exercício 01
Elabore um programa que imprima na tela a seguinte frase `Olá Mundo! Esse é o meu primeiro programa`

In [11]:
print("Olá Mundo! Esse é o meu primeiro programa")

Olá Mundo! Esse é o meu primeiro programa


## Exercício 02
Elabore um programa que escreve seu nome completo na primeira linha, seu endereço na segunda e o CEP e telefone na terceira.  

**Exemplo:**
```
Nome: Bruno Fabri
Endereço: Rua ABC
CEP: 002220-010
```

In [7]:
print("Bruno Amorim\nEndereço: Rua:Blue, 2021\nCEP:00000-000\n(11)00000-0000")


Bruno Amorim
Endereço: Rua:Blue, 2021
CEP:00000-000
(11)00000-0000


## Exercício 03
Elabore um programa que recebe o nome de uma pessoa do terminal e mostra a seguinte mensagem: `Olá {nome}! Seja bem vindo ao fantástico mundo da programação`

In [17]:
nome = input("Digite o seu nome:")
print(f"Olá {nome}! Seja bem vindo ao fantástico mundo da programação")

Digite o seu nome:Bruno
Olá Bruno! Seja bem vindo ao fantástico mundo da programação


## Exercício 04
Elabore um programa que recebe dois valores inteiros e mostra a soma desses valores

**Exemplo:**

```
Primeiro Valor = 2
Segundo Valor = 3
Soma = 5
```

In [2]:
numero1 = int(input("Digite o primeiro Valor "))
numero2 = int(input("Digite o segundo Valor "))
soma = numero1 + numero2
print(f"Soma = {soma}")

Digite o primeiro Valor 5
Digite o segundo Valor 8
Soma = 13


In [3]:
numero1 = float(input("Digite o Primeiro Valor = "))
numero2 = float(input("Digite o Segundo Valor = "))
soma = numero1 + numero2
print(f"Soma = {soma}")

Digite o Primeiro Valor = 20.3
Digite o Segundo Valor = 11.2
Soma = 31.5


## Exercício 05
Elabore um programa que recebe dois valores inteiros e mostra se o primeiro valor é maior ou igual ao segundo valor

**Exemplo:**
```
Primeiro Valor = 3
Segundo Valor = 2
Resultado = True
```

In [4]:
numero1 = float(input("Digite o primeiro Valor "))
numero2 = float(input("Digite o segundo Valor "))
print(numero1 >= numero2)

Digite o primeiro Valor 10
Digite o segundo Valor 2
True


## Exercício 06
Elabore um programa que recebe o seu nome, endereço e hobby e mostra cada uma das informações da seguinte forma:
- Nome -> Letra maiúscula
- Endereço -> Letra minúscula
- Hobby -> Primeira letra maiúscula

**Exemplo Entrada:**
```
Nome: bruno fabri
Endereço: Rua ABC
Hobby: jogar cs
```

**Exemplo Saída:**
```
Nome: BRUNO FABRI
Endereço: rua abc
Hobby: Jogar cs
```

In [8]:
nome = input("Digite seu nome: ").upper()
endereco = input("Digite seu endereco: ").lower()
hobby = input("Digite seu hobby: ").capitalize()

print("Nome: " + nome.upper())
print("Endereco: " + endereco.lower())
print("Hobby: " + hobby.capitalize())


Digite seu nome: bruno
Digite seu endereco: rua a
Digite seu hobby: desenhar
Nome: BRUNO
Endereco: rua a
Hobby: Desenhar


# Mini Projetos

## Mini Projeto 01 - Conversor de Moeda
Vamos construir um programa que irá converter moedas do real para o dólar e do dólar para o real. Vamos considerar que `$ 1,00 = R$ 5,75`

### Parte 1
Faça o conversor de moeda receber o valor em real e mostrar o valor convertido para dólar no formato `$ XXXX.XX`

**Exemplo:**

```
Valor em R$ = 1000
Valor em $ = $ 173.91
```

### Parte 2
Altere o conversor de moedas para receber o valor em dólar, converter para real e mostrar o resultado no formato `R$ XXXX.XX`

**Exemplo:**
```
Valor em $ = 1000
Valor em R$ = R$ 5750.00
```

## Mini Projeto 02 - Calculadora de aumento de aluguel
Vamos construir um programa que irá calcular o aumento anual do seu aluguel em duas partes:

### Parte 1
A sua calculadora vai receber o `valor do aluguel` e calcular o aumento baseado no `IGPM de 31%`. A calculadora deve apresentar o aluguel reajustado no formato `R$ XXXX.XX`

**Exemplo:**
```
Valor do aluguel = 1000
Valor do aluguel reajustado = R$ 1310,00
```

### Parte 2
Agora, altere sua calculadora para receber além do `valor do aluguel`, o percentual do reajuste no formato `XX%`.  

**Dica:** Descubra uma forma de transformar o percentual recebido em um número para efetuar o cálculo.

**Exemplo:**
```
Valor do aluguel = 1000
Percentual do reajuste = 31%
Valor do aluguel reajustado = R% 1310,00
```

## Mini Projeto 03 - Calculadora de dano
Vamos implementar a calculadora de dano de RPG!!

### Parte 1
O programa vai receber a `vida e um monstro (entre 10 e 50)` e o `valor do ataque do jogador por turno (entre 5 e 10)`

Baseado nos valores, exiba a quantidade de turnos que o jogador irá demorar para conseguir derrotar o monstro.

**Exemplo:**
```
Vida de um monstro (entre 10 e 50): 26
Valor do ataque do jogador por turno (entre 5 e 10): 5
Resultado: O jogador irá precisar de 6 turnos para derrotar o monstro.
```

### Parte 2
Altere o programa para ao invés de receber a vida do monstro, gerar aleatoriamente um valor entre 10 e 50.